### PR TITLE
[WIP] add CI-mode for later continuous integration: ./roots ci

### PR DESCRIPTION
**Pull request by bittorf**
The basic idea is to rely on an external script,
which builds a kernel and includes our initrd.

We have now 3 possible calls:

```
./rootfs
./rootfs chroot
./rootfs ci
```

### CODE REVIEW

**Comment by fosslinux** [comment]
All of this logic belongs in `.cirrus.yml` once my CI PR is merged. Hence, waiting on  #3.

**Comment by bittorf** [comment]
IMHO it is better to keep the affinity to `cirrus` as low a possible.
Also think about those users, which want to test locally.

`.cirrus.yml` can not be run an my computer...

**Comment by fosslinux** [comment]
In that case, then it should not be called `ci`. If we are not using it for only CI, then it needs to not be called CI in rootfs.sh.

**Comment by bittorf** [comment]
Please propose another name for Argument 1.
I need an easy way to start qemu (later also for other architectures).

**Comment by fosslinux** [comment]
1. We will not be using github actions. It does not have nested KVM support. And it is not as simple as just starting one command, we need to install dependencies, etc.
2. We already have an easy way to start qemu, `./rootfs.sh`.
3. We could use `minikernel`?

**Review by fosslinux — CHANGES_REQUESTED** [review]

File: rootfs.sh [hunk 1]
```diff
@@ -147,15 +147,31 @@ cd tmp
...
-	    -no-reboot \
-	    -kernel ../../kernel -initrd initramfs.igz -append console=ttyS0
-fi
+case "${QEMU_CMD}" in
+	chroot)
+		sudo PATH="/after/bin:${PATH}" chroot . /init
+	;;
+	ci|CI)
+		# https://github.com/bittorf/kritis-linux
+		git clone --depth 1 https://github.com/bittorf/kritis-linux.git
```
Comment (fosslinux) [inline rootfs.sh:156]: Also please checkout a specific commit here for reproducibility

**Comment by bittorf** [comment]
ok, i have applied all your suggestions.

File: rootfs.sh [hunk 2]
```diff
@@ -147,15 +147,31 @@ cd tmp
...
+
+		kritis-linux/ci_helper.sh \
+			--arch x86_64 \
+			--ramsize 1G \
+			--kernel latest \
+			--initrd "$(pwd)/initramfs.igz" \
+			--maxwait 3600 \
+			--pattern "not syncing: Attempted to kill init" \
+			--log "$(pwd)/ci.log"
```
[the remark this answers is not in the document]
Comment (fosslinux) [reply rootfs.sh]: Just this one left, please either add logging to all of them or remove logging from this one.

**Comment by fosslinux** [comment]
Please rename the commit to more accuratley reflect the changes being made, something like `Add new minikernel rootfs.sh build target` is good.

**Comment by bittorf** [comment]
i have changed to commit text.

**Comment by fosslinux** [comment]
Please rebase with master, and fix the logging problem (either standardize a log across all of them or remove the log).

**Comment by bittorf** [comment]
rebased and unified LOGFILE

**Pull request merged**